libs/platform/src/lib/components/form/input-group/input-group.component.ts

```diff
@@ -0,0 +1,227 @@
+    QueryList,
+    ElementRef,
+    ViewEncapsulation,
+    ChangeDetectorRef,
+    Renderer2,
+    AfterViewInit,
+    OnInit,
+    AfterContentInit,
+    Optional,
+    Self,
+    SkipSelf,
+    Host,
+    forwardRef,
+    ContentChild,
+    ViewChild
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { takeUntil, startWith, filter } from 'rxjs/operators';
+
+import { BaseInput } from '../base.input';
+import { FormField } from '../form-field';
+import { InputComponent } from '../input/input.component';
+import { ContentDensity, Status, FormFieldControl } from '../form-control';
+
+import { CSS_CLASS_NAME, INPUT_GROUP_CHILD_TOKEN } from './constants';
+import { InputGroupConfig } from './input-group.config';
+import { InputGroupAddonComponent } from './addon.component';
+import { InputGroupInputComponent } from './input.component';
+
+/**
+ * Fundamental input group component
+ *
+ * ```html
+ * <fdp-input-group>
+ *   <fdp-input-group-addon>$</fdp-input-group-addon>
+ *   <fdp-input-group-input type="number"></fdp-input-group-input>
+ *   <fdp-input-group-addon>0.00</fdp-input-group-addon>
+ * </fdp-input-group>
+ * ```
+ *
+ */
+@Component({
+    selector: 'fdp-input-group',
+    templateUrl: './input-group.component.html',
+    styleUrls: ['./input-group.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: forwardRef(() => InputGroupComponent), multi: true }]
+})
+export class InputGroupComponent extends BaseInput implements OnInit, AfterContentInit, AfterViewInit {
+    /**
+     * content Density of element: 'cozy' | 'compact'
+     */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+
+        this._setAddonsOptions();
+    }
+    get contentDensity(): ContentDensity {
+        return this._contentDensity;
+    }
+
+    /**
+     * control status: 'success' | 'error' | 'warning' | 'default' | 'information'
+     */
+    @Input()
+    set state(status: Status) {
+        this._controlStateClass = `is-${status}`;
+        this._lastState = status;
+    }
+
+    /** Input value */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+    /** @hidden */
+    @ContentChildren(INPUT_GROUP_CHILD_TOKEN as any)
+    _children: QueryList<InputGroupInputComponent | InputGroupAddonComponent>;
+
+    /** @hidden */
+    @ContentChild(InputGroupInputComponent)
+    _input: InputGroupInputComponent;
+
+    /** @hidden */
+    @ViewChild(InputComponent, { read: ElementRef })
+    set _inputComponentElement(inputComponentElementRef: ElementRef<HTMLElement>) {
+        this._setInputElementClass(inputComponentElementRef?.nativeElement);
+    }
+
+    /** @hidden */
+    _beforeInputAddons: InputGroupAddonComponent[] = [];
```

Comment:
We will have to update our current coding style guide as we have some inconsistencies with the private/non-private properties. 

We need to allow `_xxxx` also for public members which are part of template, but belongs to private API.

Reply:
@fkolar the corresponding [issue](https://github.com/SAP/fundamental-ngx/issues/3211) has been created